components/api/src/main/java/com/hotels/styx/api/HttpRequest.java

```diff
@@ -362,14 +350,40 @@ public Builder newBuilder() {
         }
     }
 
+    /**
+     * Decodes the "Cookie" header in this request and returns the cookies.
+     *
+     * @return cookies
+     */
+    public Set<RequestCookie> cookies() {
+        // Note: there should only be one "Cookie" header, but we check for multiples just in case
+        // the alternative would be to respond with a 400 Bad Request status if multiple "Cookie" headers were detected
```

Comment:
The RFC clearly states that there should be a single Cookie header. I think we can return an error in that case.

Reply:
We would need to return the error in the builder. The `cookies` method could be changed to assume only one `Cookie` header exists though.